src/org/zaproxy/zap/extension/viewstate/HttpPanelViewStateView.java

```diff
@@ -0,0 +1,268 @@
+			// Setup text field for ViewState info
+			vsInfoTxt = new JFormattedTextField();
+			vsInfoTxt.setEditable(false);
+			vsInfoTxt.setBackground(Color.decode("#D6D9DF"));
+			// Add to container
+			container.add(cScroll);
+			container.add(vsInfoTxt);
+		}
+		return container;
+	}
+
+	private JTable getHexTableBody() {
+		if (hexTableBody == null) {
+			hexTableBody = new JTable();
+			hexTableBody.setName("");
+			hexTableBody.setModel(getHttpPanelHexModel());
+
+			hexTableBody.setGridColor(java.awt.Color.gray);
+			hexTableBody.setIntercellSpacing(new java.awt.Dimension(1,1));
+			hexTableBody.setRowHeight(DisplayUtils.getScaledSize(18));
+			
+			hexTableBody.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+			hexTableBody.getColumnModel().getColumn(0).setPreferredWidth(100);
+			for (int i=1; i <= 17; i++) {
+				hexTableBody.getColumnModel().getColumn(i).setPreferredWidth(30);
+			}
+			for (int i=17; i <= hexTableBody.getColumnModel().getColumnCount()-1; i++) {
+				hexTableBody.getColumnModel().getColumn(i).setPreferredWidth(25);
+			}
+			
+			hexTableBody.setCellSelectionEnabled(true);
+			hexTableBody.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+		}
+		return hexTableBody;
+	}
+
+	public HttpPanelViewStateModel getHttpPanelHexModel() {
+		if (httpPanelHexModel == null) {
+			httpPanelHexModel = new HttpPanelViewStateModel();
+		}
+		return httpPanelHexModel;
+	}
+	
+	public void setEnabled(boolean enabled) {
+		isEnabled = enabled;
+	}
+
+	@Override
+	public boolean isEnabled(Message aMessage) {
+		return isEnabled;
+	}
+
+	@Override
+	public boolean hasChanged() {
+		return getHttpPanelHexModel().hasChanged();
+	}
+
+	@Override
+	public boolean isEditable() {
+		return isEditable;
+	}
+
+	@Override
+	public void setEditable(boolean editable) {
+		getHttpPanelHexModel().setEditable(editable);
+		if (!editable) {
+			getHttpPanelHexModel().setData(new byte[0]);
+		}
+	}
+
+	@Override
+	public void dataChanged(HttpPanelViewModelEvent e) {
+		getHttpPanelHexModel().setData( model.getData() );
+	}
+	
+	@Override
+	public void save() {
+		model.setData(getHttpPanelHexModel().getData());
+	}
+	
+	@Override
+	public void setParentConfigurationKey(String configurationKey) {
+	}
+	
+	@Override
+	public void loadConfiguration(FileConfiguration fileConfiguration) {
+	}
+	
+	@Override
+	public void saveConfiguration(FileConfiguration fileConfiguration) {
+	}
+	
+	@Override
+	public void setSelected(boolean selected) {
+		if (selected) {
+			hexTableBody.requestFocusInWindow();
+		}
+	}
+	
+	@Override
+	public HttpPanelViewModel getModel() {
+		return model;
+	}
+
+	@Override
+	public void viewStateUpdated(ViewState vs) {
+		if (vs != null) {
+			String info = "";
+			logger.info("ViewState updated: " + vs.getType() + " :: " + vs.getValue());
+			vsInfoTxt.setFont(new Font("Courier", Font.BOLD, 12));
+			if (vs.getType().equalsIgnoreCase("JSF")) {
+				if (vs.getDecodedValue() != null) {
+					info = "Type: " + vs.getType();
```

Comment:
Add the strings to the Messages.properties file and use the method `Constant.messages.getString(String)` to obtain them.